File: 0078-subsets/0078-subsets.java
class Solution {
    List<Integer> path = new LinkedList<>();
    List<List<Integer>> ans = new ArrayList<>();
    
    public List<List<Integer>> subsets(int[] nums) {
        backtracking(nums, 0);
        return ans;
    }
    
    private void backtracking(int[] nums, int startIdx) {
        ans.add(new ArrayList<>(path));
        
        for (int i = startIdx; i < nums.length; i++) {
            path.add(nums[i]);
            backtracking(nums, i + 1);
            path.removeLast();
        }
    }
}

File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []

        def dfs(start: int, path: list[int]) -> None:
            res.append(path[:])

            for i in range(start, len(nums)):
                dfs(i + 1, path + [nums[i]])
        
        dfs(0, [])
        return res